master/internal/stream/system.go

```diff
@@ -367,36 +384,37 @@ func startup[T stream.Msg, S any](
 	state.Subscription.Configure(filter)
 
 	// Scan for historical msgs matching newly-added subscriptions.
-	var newmsgs []*websocket.PreparedMessage
-	newmsgs, err = state.CollectStartupMsgs(known, *spec, ctx)
+	var newmsgs []interface{}
+	newmsgs, err = state.CollectStartupMsgs(known, *spec, ctx, state.Subscription.UpsertFunc, state.Subscription.DeleteFunc)
 	if err != nil {
 		return nil, err
 	}
 	msgs = append(msgs, newmsgs...)
 	return msgs, nil
 }
 
+// is startup a verb or noun?
```

Comment:
It's a gerund (or a noun with side effects), it's responsible for handling the initial setup of a new subscription/streamer, while also getting the most recent messages that meet the subscription's filter